ads-api/src/main/java/io/voodoo/apps/ads/api/AdClient.kt

```diff
@@ -284,17 +287,22 @@ abstract class BaseAdClient<ActualType : PublicType, PublicType : Ad>(
         adClickListeners.remove(listener)
     }
 
-    override fun destroyAdsIf(predicate: (Ad) -> Boolean) {
+    /**
+     * @return number of elements being destroyed
+     */
+    override fun destroyAdsIf(predicate: (Ad) -> Boolean) : Int {
         synchronized(loadedAds) {
-            val renderedAds = loadedAds.filter(predicate)
+            val adsToDestroy = loadedAds.filter(predicate)
 
-            val renderedAdsNotLocked = loadedAds.filter { it.isLocked().not() }
+            val adsToDestroyNotLocked = loadedAds.filter { it.isLocked().not() }
             pendingAdsToRelease.addAll(loadedAds.filter { it.isLocked() }.map { it.id })
 
-            if (renderedAdsNotLocked.isEmpty()) return
+            if (adsToDestroyNotLocked.isEmpty()) return adsToDestroy.size
```

Comment:
whatever they are locked or not, the size should be `adsToDestroy.size` not `adsToDestroyNotLocked.size`

Reply:
yup agree